flink-python/src/main/java/org/apache/flink/streaming/api/operators/python/StatefulPythonFunctionOperator.java

```diff
@@ -0,0 +1,237 @@
+		super.open();
+		bais = new ByteArrayInputStreamWithPos();
+		baisWrapper = new DataInputViewStreamWrapper(bais);
+
+		baos = new ByteArrayOutputStreamWithPos();
+		baosWrapper = new DataOutputViewStreamWrapper(baos);
+		runnerInputSerializer = PythonTypeUtils.TypeInfoToSerializerConverter
+			.typeInfoSerializerConverter(runnerInputTypeInfo);
+		runnerOutputSerializer = PythonTypeUtils.TypeInfoToSerializerConverter
+			.typeInfoSerializerConverter(runnerOutputTypeInfo);
+
+		InternalTimerService<VoidNamespace> internalTimerService = getInternalTimerService("user-timers",
+			VoidNamespaceSerializer.INSTANCE, this);
+
+		this.streamRecordCollector = new StreamRecordCollector(output);
+		timerservice = new SimpleTimerService(internalTimerService);
+		reusableInput = new Row(5);
+		reusableTimerData = new Row(5);
+	}
+
+	@Override
+	public TypeInformation<OUT> getProducedType() {
+		return outputTypeInfo;
+	}
+
+	@Override
+	public void onEventTime(InternalTimer<Row, VoidNamespace> timer) throws Exception {
+		processTimer(false, timer);
+		checkInvokeFinishBundleByCount();
+	}
+
+	@Override
+	public void onProcessingTime(InternalTimer<Row, VoidNamespace> timer) throws Exception {
+		processTimer(true, timer);
+		checkInvokeFinishBundleByCount();
+	}
+
+	@Override
+	public PythonFunctionRunner createPythonFunctionRunner() throws Exception {
+		return new BeamDataStreamStatefulPythonFunctionRunner(
+			getRuntimeContext().getTaskName(),
+			createPythonEnvironmentManager(),
+			runnerInputTypeInfo,
+			runnerOutputTypeInfo,
+			DATA_STREAM_STATEFUL_PYTHON_FUNCTION_URN,
+			PythonOperatorUtils.getUserDefinedDataStreamStatefulFunctionProto(
+				pythonFunctionInfo, getRuntimeContext(), Collections.EMPTY_MAP, keyTypeInfo),
+			DATA_STREAM_STATEFUL_MAP_FUNCTION_CODER_URN,
+			jobOptions,
+			getFlinkMetricContainer(),
+			getKeyedStateBackend(),
+			keyTypeSerializer,
+			getContainingTask().getEnvironment().getMemoryManager(),
+			getOperatorConfig().getManagedMemoryFractionOperatorUseCaseOfSlot(
+				ManagedMemoryUseCase.PYTHON,
+				getContainingTask().getEnvironment().getTaskManagerInfo().getConfiguration(),
+				getContainingTask().getEnvironment().getUserCodeClassLoader().asClassLoader()));
+	}
+
+	@Override
+	public PythonEnv getPythonEnv() {
+		return this.pythonFunctionInfo.getPythonFunction().getPythonEnv();
+	}
+
+	@Override
+	public void emitResult(Tuple2<byte[], Integer> resultTuple) throws Exception {
+		byte[] rawResult = resultTuple.f0;
+		int length = resultTuple.f1;
+		bais.setBuffer(rawResult, 0, length);
+		Row runnerOutput = (Row) runnerOutputSerializer.deserialize(baisWrapper);
+		if (runnerOutput.getField(0) != null) {
+			registerTimer(runnerOutput);
+		} else {
+			streamRecordCollector.collect(runnerOutput.getField(3));
+		}
+
+	}
+
+	@Override
+	public void processElement(StreamRecord<Row> element) throws Exception {
+		LOGGER.info("Current watermark: " + timerservice.currentWatermark());
```

Comment:
Remove this logging. It will output a log for each input element.

Reply:
Sorry, it is for debug purpose and should be removed.